### PR TITLE
Chartbeat integration

### LINKED ISSUE #1987: Integrate Chartbeat
**Is your feature request related to a problem? Please describe.**
Following on from the completion of #1986, we should now integrate Chartbeat analytics into Simorgh. The table provided in the above issue contains the relevant information that we need to pass to Chartbeat.

**Describe the solution you'd like**
Chartbeat should work with both AMP and canonical articles. Using the utilities created in #1986, we should now be able to send a beacon to Chartbeat with the relevant data.

You can check that requests are being sent to Chartbeat by observing your network requests panel and looking for the domain `ping.chartbeat.com`.

As a result of this network request being made, you should then see real-time stats in your Chartbeat dashboard.

For any environment other than live, we should report to the domain `test.bbc.co.uk`.

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Unit + Cypress tests

**Additional context**
Chartbeat docs: http://support.chartbeat.com/docs/
Chartbeat for AMP: http://support.chartbeat.com/docs/integrations.html#amp

Regarding single page applications, [Chartbeat docs](https://chartbeat.zendesk.com/hc/en-us/articles/360017557814-Integrate-Chartbeat-on-your-website) say:

> If your website makes use of dynamic page change, infinite scroll, or Single Page App architecture, you will need to implement our virtualPage function for Chartbeat to properly track these pageviews.

However, the [virtualPage documentation](http://support.chartbeat.com/docs/#ajax) states that it _should_ only be needed if the DOM does not update and/or the URL does not change. Therefore, I think we should be okay, but might need a little bit of investigation.
 
Chartbeat recommend placing the code to fire the beacon within the `head` of the page, and after the `rel="canonical"`.

I'd also recommend installing the [Chartbeat Heads-Up Display](https://chartbeat.zendesk.com/hc/en-us/articles/115001293233-Install-the-Heads-Up-Display-HUD-)

If you need Chartbeat access, speak to @jimjohnsonrollings

### PR DESCRIPTION
**Pull request by rhenshaw56**
Resolves #1987

**Overall change:** Integrated Chartbeat Analytics

**Code changes:**
  - added charbeat beacon to head tag
  - created beacon and initialization code for amp and canonical pages

**To test**
  - You can check that requests are being sent to Chartbeat by observing your network requests panel and looking for the domain `ping.chartbeat.com`.
  - Install [Chartbeat Heads-Up Display](https://chartbeat.zendesk.com/hc/en-us/articles/115001293233-Install-the-Heads-Up-Display-HUD-) and observe that it fires on the page

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: src/app/containers/ChartbeatAnalytics/canonical/index.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import {
+  chartbeatUID,
+  useCanonical,
+  getSylphidCookie,
+} from '../../../lib/analyticsUtils/chartbeat';
+
+const domain = 'test.bbc.co.uk';
+
+const inlinedScriptSnippet = () => `
```
Comment (12) [inline src/app/containers/ChartbeatAnalytics/canonical/index.jsx]: I know this is still in progress, but just wanted to add that this needs to be added to the `head` of the page. You can do this with `react-helmet`.

**Review by RayNjeri — APPROVED** [review]
LGTM

File: src/app/containers/ArticleMain/index.jsx [hunk 2]
```diff
@@ -9,6 +9,7 @@ import Blocks from '../Blocks';
 import timestamp from '../ArticleTimestamp';
 import { GhostGrid } from '../../lib/styledGrid';
 import ATIAnalytics from '../ATIAnalytics';
+import ChartBeatAnalytics from '../ChartbeatAnalytics';
```
Comment (12) [inline src/app/containers/ArticleMain/index.jsx]:
```suggestion
import ChartbeatAnalytics from '../ChartbeatAnalytics';
```

File: src/app/containers/ArticleMain/index.jsx [hunk 3]
```diff
@@ -51,6 +52,7 @@ const ArticleMain = ({ articleData }) => {
   return (
     <Fragment>
       <ATIAnalytics data={articleData} />
+      <ChartBeatAnalytics />
```
Comment (12) [inline src/app/containers/ArticleMain/index.jsx]:
```suggestion
      <ChartbeatAnalytics />
```

File: src/app/containers/ChartbeatAnalytics/canonical/__snapshots__/index.test.jsx.snap [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`CanonicalChartbeatAnalytics should return script snippet 1`] = `null`;
```
Comment (12) [inline src/app/containers/ChartbeatAnalytics/canonical/__snapshots__/index.test.jsx.snap:3]: Is this snapshot correct?

File: src/app/containers/ChartbeatAnalytics/canonical/index.jsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+          };
+          function loadChartbeat() {
+           var e = document.createElement('script');
+           var n = document.getElementsByTagName('script')[0];
+           e.type = 'text/javascript';
+           e.async = true;
+           e.src = '//static.chartbeat.com/js/chartbeat.js';
+           n.parentNode.insertBefore(e, n);
+          }
+          loadChartbeat();
```
Comment (dr3) [inline src/app/containers/ChartbeatAnalytics/canonical/index.jsx:35]: While the chartbeat script itself if async, this code runs synchronously, blocking paint. Could this be improved? Perhaps making it run async, or stopping this running server side, so it only loads during the bundle hydration?

File: src/app/containers/ChartbeatAnalytics/index.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    chartbeatUID,
+  };
+
+  const beacon =
+    platform === 'amp' ? (
+      <AmpChartbeatsBeacon {...commonProps} />
+    ) : (
+      <CanonicalChartbeatsBeacon {...commonProps} useCanonical={useCanonical} />
+    );
+  return beacon;
```
Comment (12) [inline src/app/containers/ChartbeatAnalytics/index.jsx]:
```suggestion
```

File: src/app/containers/ChartbeatAnalytics/index.jsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  const type = getType(pageType);
+  const commonProps = {
+    domain,
+    type,
+    sections,
+    cookie,
+    chartbeatUID,
+  };
+
+  const beacon =
```
Comment (12) [inline src/app/containers/ChartbeatAnalytics/index.jsx]:
```suggestion
  return
```

**Comment by 12** [comment]
Can confirm that when checked out locally, this is working as expected (from what I can tell), with sections being generated (though producer + chapter values do not exist yet). Chartbeat HUD is working and reporting to `test.bbc.co.uk`.

File: src/app/containers/ChartbeatAnalytics/canonical/__snapshots__/index.test.jsx.snap (hunk 4 above)
Comment (rhenshaw56) [reply src/app/containers/ChartbeatAnalytics/canonical/__snapshots__/index.test.jsx.snap:3]: yeah, apparently that's what it returns which is weird. similar to [this](https://github.com/bbc/simorgh/blob/latest/src/app/components/AudioVideoHead/__snapshots__/index.test.jsx.snap#L3)

File: src/app/containers/ChartbeatAnalytics/canonical/index.jsx (hunk 5 above)
Comment (rhenshaw56) [reply src/app/containers/ChartbeatAnalytics/canonical/index.jsx:35]: Thanks @dr3, I have added the `async` attribute to the script tag

File: src/app/containers/ChartbeatAnalytics/amp/index.jsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    sections: options.sections,
+    domain: options.domain,
+    contentType: options.type,
+    idSync: {
+      bbc_hid: options.cookie,
+    },
+  },
+});
+
+const JsonInlinedScript = data => (
```
Comment (12) [inline src/app/containers/ChartbeatAnalytics/amp/index.jsx:16]: We're trying to move away from having to use `dangerouslySetInnerHTML`, can we look into taking the approach used in the original AV implementation? 

https://github.com/bbc/simorgh/blob/0df6465e69241433dabac1a0be71a757380048e1/src/app/components/AudioVideoHead/index.jsx#L23-L47

Comment (rhenshaw56) [reply src/app/containers/ChartbeatAnalytics/amp/index.jsx:16]: Thanks @rhell, I tried that earlier but I don't think that will work with amp, the only acceptable types for the `script` tag for amp would be `"application/json"` and `"text/plain"` [amp docs](https://amp.dev/documentation/guides-and-tutorials/learn/spec/amphtml/#html-tags)

**Comment by rhenshaw56** [comment]
closing this PR as https://github.com/bbc/simorgh/pull/2665 uses this implementation

**Pull request closed without merging**